### PR TITLE
Fixed goalGradeSetByUser type

### PR DESCRIPTION
**Pull request by justin0022**
Resolves https://github.com/tl-its-umich-edu/my-learning-analytics/issues/953.

Instead of `null` as the default value, I've switched it to `false`.

### CODE REVIEW

**Comment by ssciolla** [comment]
Hi Justin, I haven’t tried running this yet, but I think maybe I didn’t communicate the issue clear enough. I’m seeing sometimes goals (e.g. 92.5) stored in `goalGradeSetByUser`, but it seems like this should always have a Boolean value, right? I can follow up with more details tomorrow. Thanks!

File: assets/src/util/assignment.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ const setAssignmentGoalGrade = (key, assignments, goalGrade) => {
     {
       ...assignments[key],
       goalGrade: goalGrade === '' ? '' : roundToXDecimals(Number(goalGrade), 3),
-      goalGradeSetByUser: goalGrade
+      goalGradeSetByUser: !!goalGrade
```
Comment (justin0022) [inline assets/src/util/assignment.js:29]: Sam I think this is the change that should fix this issue? Sorry I think I may have incorrectly merged the master in the previous commit - this one should work.

Comment (ssciolla) [reply assets/src/util/assignment.js:29]: Cool, thanks, I'll try this in the next couple hours and review.

**Comment by ssciolla** [comment]
@pushyamig, I'm reviewing this, but I wanted you to be aware of the issue/changes.

**Comment by pushyamig** [comment]
@ssciolla OK I will also look into it

**Comment by justin0022** [comment]
> in my testing, I never saw a case where `goalGradeSetByUser` was stored as `false`. 

If you uncheck the "Lock Goal", the associated assignment `goalGradeSetByUser` should be set to `false`. I use this as a way to filter for `gradedAssignments` and `ungradedAssignments`. 

The idea is that `gradedAssignments` can be considered as assignments that either are graded or the goal grade is set by the user, which are the "fixed" grades, and that `ungradedAssignments` are the inverse, which are the "unfixed" grades that can be calculated to match the overall course goal grade.

**Comment by ssciolla** [comment]
> If you uncheck the "Lock Goal", the associated assignment `goalGradeSetByUser` should be set to `false`. I use this as a way to filter for `gradedAssignments` and `ungradedAssignments`.

Okay, but I'm when I uncheck "Lock Goal", I'm not seeing anything stored at all for that assignment.

With one goal, it looks like this: 

``` 
{"goalGrade": "90", "assignments": [{"assignmentId": "1770000000XXXXXX", "goalGradeSetByUser": true, "goalGrade": 9}]}
```

And after I un-check, it looks like this:

```
{"goalGrade": "90", "assignments": []}
```

This doesn't mean it's not working, I'm just trying to understand the logic of what's stored in the DB.

**Comment by justin0022** [comment]
> Okay, but when I uncheck "Lock Goal", I'm not seeing anything stored at all for that assignment.

The user settings are not set unless "Lock Goal" is checked, since when this is unchecked, it means that the grade calculation is done automatically by the application. 

The added benefit is that by not saving the automatically calculated grades, if there are additional assignments added to the course in the future, the calculated grades can freely adjust. If we stored these grades, and new assignments get added, then the old stored grades will need to be discarded, recalculated, and stored. 

If there's a need to store the calculated grades, it can definitely be done!

**Comment by ssciolla** [comment]
> The user settings are not set unless "Lock Goal" is checked, since when this is unchecked, it means that the grade calculation is done automatically by the application.

Not storing the system-generated (un-locked) goals is fine with me. I was responding to your earlier assertion that "If you uncheck the "Lock Goal", the associated assignment `goalGradeSetByUser` should be set to `false`." I have yet to see `false` be stored in the database. Maybe you meant that because it's undefined, it's `falsy` and the system knows that.  But then you get back to my question of why this field is needed in the first place. You probably could just check whether `goalGrade` is set rather than I relying on this boolean to decide whether it should be included in `gradedAssignments`. I understand the convenience of having a separate flag, and to be clear, I'm not asking you to change it.

**Comment by justin0022** [comment]
Sorry I meant it was stored on the front end in the AssignmentTable component - not in the back-end. There was a reason I had to store `goalGradeSetByUser` independently from `checkboxState` - I can't exactly recall now why I added it, but I do remember thinking it was too bad that this additional state was necessary. 

If everything works as expected without `goalGradeSetByUser`, I'm all for removing it.

**Comment by ssciolla** [comment]
Thanks for the explanation, @justin0022. If you want to explore removing it, I'm okay with that, but I'm also totally fine with keeping it as is for now. As noted elsewhere, we are trying to release this week, so merging and creating a separate issue would be a fine way to proceed.

**Comment by justin0022** [comment]
Thanks I'll keep it on my backlog, I'll merge this in for now.

**Pull request merged**